rest/config.go

```diff
@@ -1023,9 +1023,22 @@ func RunServer(config *ServerConfig) {
 	}
 
 	sc := NewServerContext(config)
-	for _, dbConfig := range config.Databases {
+	bucketUUIDToDbs := make(map[string][]string, len(config.Databases))
+
+	for dbName, dbConfig := range config.Databases {
 		if _, err := sc.AddDatabaseFromConfig(dbConfig); err != nil {
-			base.Fatalf("Error opening database %s: %+v", base.MD(dbConfig.Name), err)
+			base.Fatalf("Error opening database %s: %+v", base.MD(dbName), err)
+		}
+		dbContext, err := sc.GetDatabase(dbName)
+		if err != nil {
+			base.Fatalf("Error getting database context %s: %+v", base.MD(dbName), err)
+		}
+		if uuid, err := dbContext.Bucket.UUID(); err == nil {
+			if value := bucketUUIDToDbs[uuid]; value != nil {
+				base.Warnf("Unsupported config: Database %q shares bucket %q with databases %s",
+					base.MD(dbName), base.MD(dbContext.Bucket.GetName()), base.MD(value))
+			}
+			bucketUUIDToDbs[uuid] = append(bucketUUIDToDbs[uuid], dbName)
```

Comment:
Can you add a unit test for this functionality, to ensure that all scenarios behave as expected?

Reply:
Of course; added a unit test, TestValidateAndAddDBFromConfig and introduced validateAndAddDBFromConfig function to make the logic testable. Changes are available in latest revision.